cli/src/main/java/com/box/l10n/mojito/cli/command/CheckerCommand.java

```diff
@@ -0,0 +1,194 @@
+        if(failedCheckNames.size() > 0) {
+            consoleWriter.fg(Ansi.Color.YELLOW).newLine().a("Failed checks: ").println();
+            failedCheckNames.stream().forEach(check -> consoleWriter.fg(Ansi.Color.YELLOW).newLine().a("\t* " + check).println());
+            consoleWriter.fg(Ansi.Color.YELLOW).newLine().a("Sending notifications.").println();
+            //TODO: Checks failed, send notifications
+        }
+        consoleWriter.fg(Ansi.Color.GREEN).newLine().a("Checks completed").println(2);
+    }
+
+    private CliCheckerOptions generateCheckerOptions() {
+        Set<String> regexSet = generateParameterRegexSet();
+        Set<String> hardFailureSet = generateHardFailureSet();
+        return new CliCheckerOptions(regexSet, hardFailureSet, dictionaryAdditionsFilePath, glossaryFilePath);
+    }
+
+    private Set<String> generateHardFailureSet() {
+        Set<String> hardFailureSet = new HashSet<>();
+        Stream<CliCheckerType> cliCheckerTypeStream = Arrays.stream(CliCheckerType.values());
+        if (hardFailList != null) {
+            if (hardFailList.equalsIgnoreCase("all")) {
+                cliCheckerTypeStream.forEach(cliCheckerType -> hardFailureSet.add(cliCheckerType.getClassName()));
+            } else {
+                String[] hardFails = hardFailList.split(",");
+                for (String checkName : hardFails) {
+                    if (cliCheckerTypeStream.anyMatch(cliCheckerType -> cliCheckerType.name().equals(checkName))) {
+                        hardFailureSet.add(CliCheckerType.valueOf(checkName).getClassName());
+                    } else {
+                        throw new CommandException("Unknown check name in hard fail list '" + checkName + "'");
+                    }
+                }
+            }
+        }
+        return hardFailureSet;
+    }
+
+    private Set<String> generateParameterRegexSet() {
+        Set<String> regexSet = new HashSet<>();
+        if(parameterRegexList != null && !parameterRegexList.isEmpty()) {
+            String[] regexNames = parameterRegexList.split(",");
+            Map<String, String> placeholderRegexMap = getPlaceholderRegexStringMap();
+            for(String regexName : regexNames) {
+                if (!placeholderRegexMap.containsKey(regexName)){
+                    throw new CommandException("Unknown parameter regex name " + regexName);
+                }
+                regexSet.add(placeholderRegexMap.get(regexName));
+            }
+        }
+        return regexSet;
+    }
+
+    private List<CliChecker> generateChecks(List<AssetExtractionDiff> assetExtractionDiffs) {
+        List<CliChecker> cliCheckers = new ArrayList<>();
+        String[] checks = checkerList.split(",");
+        CliCheckerOptions options = generateCheckerOptions();
+        Stream<CliCheckerType> checkerTypes = Arrays.stream(CliCheckerType.values());
+        for(String check : checks) {
+            if(checkerTypes.anyMatch(c -> c.name().equals(check))){
```

Comment:
having something like `Stream.of(checks).map(c -> createInstance...)`? Not sure why we have a double check with `checkerTypes` 

`checkerTypes.anyMatch()` calling twice `anyMatch()` on a stream will fail no?